workflows/bactmap.nf

```diff
@@ -0,0 +1,234 @@
+////////////////////////////////////////////////////
+
+ch_multiqc_config        = file("$projectDir/assets/multiqc_config.yaml", checkIfExists: true)
+ch_multiqc_custom_config = params.multiqc_config ? Channel.fromPath(params.multiqc_config) : Channel.empty()
+
+////////////////////////////////////////////////////
+/* --       IMPORT MODULES / SUBWORKFLOWS      -- */
+////////////////////////////////////////////////////
+
+// Don't overwrite global params.modules, create a copy instead and use that within the main script.
+def modules = params.modules.clone()
+
+def multiqc_options   = modules['multiqc']
+multiqc_options.args += params.multiqc_title ? Utils.joinModuleArgs(["--title \"$params.multiqc_title\""]) : ''
+
+// Local: Modules
+include { GET_SOFTWARE_VERSIONS } from '../modules/local/get_software_versions'     addParams( options: [publish_files : ['csv':'']] )
+include { VCF2PSEUDOGENOME      } from '../modules/local/vcf2pseudogenome'          addParams( options: modules['vcf2pseudogenome'])
+include { ALIGNPSEUDOGENOMES } from '../modules/local/alignpseudogenomes'           addParams( options: modules['alignpseudogenomes'])
+include { MULTIQC               } from '../modules/local/multiqc'                   addParams( options: multiqc_options )
+
+// nf-core: Modules
+include { GUBBINS } from '../modules/nf-core/software/gubbins/main'                 addParams( options: modules['gubbins'])
+include { SNPSITES } from '../modules/nf-core/software/snpsites/main'               addParams( options: modules['snpsites'])
+
+// Local: Sub-workflows
+include { INPUT_CHECK       } from '../subworkflows/input_check'       addParams( options: [:] )
+include { BAM_SORT_SAMTOOLS } from '../subworkflows/bam_sort_samtools' addParams( samtools_sort_options: modules['samtools_sort'], samtools_index_options : modules['samtools_index'], bam_stats_options: modules['bam_stats'])
+
+include { VARIANTS_BCFTOOLS } from '../subworkflows/variants_bcftools' addParams( bcftools_mpileup_options: modules['bcftools_mpileup'], bcftools_filter_options: modules['bcftools_filter'])
+include { SUB_SAMPLING } from '../subworkflows/sub_sampling'           addParams( mash_sketch_options: modules['mash_sketch'], rasusa_options: modules['rasusa'])
+
+include { CREATE_PHYLOGENY } from '../subworkflows/create_phylogeny'   addParams(   rapidnj_options: modules['rapidnj'],
+                                                                                                fasttree_options: modules['fasttree'], 
+                                                                                                iqtree_options: modules['iqtree'], 
+                                                                                                raxmlng_options: modules['raxmlng']
+                                                                                            )
+
+include { find_genome_size } from '../modules/local/functions.nf'
```

Comment:
This function can now be put in the new `lib/WorkflowBactmap.groovy` file which is there for exactly this reason. Will be added in https://github.com/nf-core/bactmap/pull/38. See [here](https://github.com/nf-core/rnaseq/blob/master/lib/WorkflowRnaseq.groovy) for other examples.